File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/MovieBookingSystem.py
import datetime

class MovieBookingSystem:
    def __init__(self):
        self.movies = []

    def addMovie(self, name, price, startTime, endTime, n):
        movie = {
            "name": name,
            "price": price,
            "start_time": datetime.datetime.strptime(startTime, "%H:%M").time(),
            "end_time": datetime.datetime.strptime(endTime, "%H:%M").time(),
            "seats": [[0] * n for _ in range(n)]
        }
        self.movies.append(movie)

    def bookTicket(self, name, seatsToBook):
        for movie in self.movies:
            if movie["name"] == name:
                seats = movie["seats"]
                for seat in seatsToBook:
                    row, col = seat
                    if seats[row][col] == 0:
                        seats[row][col] = 1
                    else:
                        return "Booking failed."
                return "Booking success."
        return "Movie not found."

    def availableMovies(self, startTime, endTime):
        start = datetime.datetime.strptime(startTime, "%H:%M").time()
        end = datetime.datetime.strptime(endTime, "%H:%M").time()
        result = []
        for movie in self.movies:
            movie_start = movie["start_time"]
            movie_end = movie["end_time"]
            if movie_start >= start and movie_end <= end:
                result.append(movie["name"])
        return result

    def getMovies(self):
        return self.movies

File: data/derived_cot/rq1_traces/translation/glm5.1/java_to_cpp/code_output/MovieBookingSystem.cpp
#ifndef MOVIE_BOOKING_SYSTEM_H
#define MOVIE_BOOKING_SYSTEM_H

#include <string>
#include <vector>
#include <map>
#include <any>
#include <algorithm>

struct LocalTime {
    int hour;
    int minute;

    LocalTime() : hour(0), minute(0) {}
    LocalTime(int h, int m) : hour(h), minute(m) {}

    static LocalTime parse(const std::string& timeStr) {
        size_t colonPos = timeStr.find(':');
        int h = std::stoi(timeStr.substr(0, colonPos));
        int m = std::stoi(timeStr.substr(colonPos + 1));
        return LocalTime(h, m);
    }

    bool isBefore(const LocalTime& other) const {
        if (hour < other.hour) return true;
        if (hour > other.hour) return false;
        return minute < other.minute;
    }

    bool isAfter(const LocalTime& other) const {
        return other.isBefore(*this);
    }
};

class MovieBookingSystem {
private:
    std::vector<std::map<std::string, std::any>> movies;

public:
    MovieBookingSystem() = default;

    void addMovie(const std::string& name, double price,
                  const std::string& startTime, const std::string& endTime, int n) {
        std::map<std::string, std::any> movie;
        movie["name"] = name;
        movie["price"] = price;
        movie["start_time"] = LocalTime::parse(startTime);
        movie["end_time"] = LocalTime::parse(endTime);
        movie["seats"] = std::vector<std::vector<int>>(n, std::vector<int>(n, 0));
        movies.push_back(movie);
    }

    std::string bookTicket(const std::string& name,
                           const std::vector<std::vector<int>>& seatsToBook) {
        for (auto& movie : movies) {
            if (std::any_cast<const std::string&>(movie.at("name")) == name) {
                auto& seats =
                    std::any_cast<std::vector<std::vector<int>>&>(movie.at("seats"));
                for (const auto& seat : seatsToBook) {
                    int row = seat[0];
                    int col = seat[1];
                    if (seats[row][col] == 0) {
                        seats[row][col] = 1;
                    } else {
                        return "Booking failed.";
                    }
                }
                return "Booking success.";
            }
        }
        return "Movie not found.";
    }

    std::vector<std::string> availableMovies(const std::string& startTime,
                                             const std::string& endTime) {
        LocalTime start = LocalTime::parse(startTime);
        LocalTime end = LocalTime::parse(endTime);

        std::vector<std::string> result;
        for (const auto& movie : movies) {
            const LocalTime& movieStart =
                std::any_cast<const LocalTime&>(movie.at("start_time"));
            const LocalTime& movieEnd =
                std::any_cast<const LocalTime&>(movie.at("end_time"));
            if (!movieStart.isBefore(start) && !movieEnd.isAfter(end)) {
                result.push_back(
                    std::any_cast<const std::string&>(movie.at("name")));
            }
        }
        return result;
    }

    std::vector<std::map<std::string, std::any>>& getMovies() {
        return movies;
    }
};

#endif